### PR TITLE
Coordinator auctions

### PR DESCRIPTION
**Pull request by JChanceHud**
Implements bidding logic in the coordinator codebase.

Refactors/optimizes burn auction contract; adds `multiBid` function.

Implements http transaction forwarding to the current block producer.

Allows many host/port combinations to be stored for a single coordinator in the burn auction with the format `host.com:8888,host2.com:9999`.

Adds Auction info menu to set max bid, custom url and refund balance.

Refactors docker setup to poll for postgres availability instead of waiting.

### CODE REVIEW

File: packages/contracts/src/tx-util.ts [hunk 1]
```diff
@@ -13,49 +11,23 @@ export class TxUtil {
...
-      nonce: NumString.from(`${nonce}`)
-        .toBytes()
-        .toString(),
-      gasPrice: NumString.from(`${gasPrice}`)
-        .toBytes()
-        .toString(),
-      gasLimit: NumString.from(`${gas}`)
-        .toBytes()
-        .toString(),
+    const [ gas, gasPrice, nonce ] = await Promise.all([
```
Comment (JChanceHud) [inline packages/contracts/src/tx-util.ts:14]: @wanseob I refactored this when I was debugging something unrelated. Let me know if you're okay with it, using web3 for tx signing is much more clear to me. 🙂

Comment (wanseob) [reply packages/contracts/src/tx-util.ts:21]: This refactoring looks very good to me, I was trying to use `web3.eth.account.ssignTransaction` but didn't notice how to pass the private key into that function, if it does also work well I like this refactoring a lot :+1:  And thanks for prettifying the async things

**Comment by JChanceHud** [comment]
@wanseob this PR should be ready to go. It doesn’t include request forwarding but that might be better in a separate PR as this one is pretty big.

**Comment by wanseob** [comment]
Thank you @JChanceHud I'll take a look at this PR :+1:

**Comment by wanseob** [comment]
image
Maybe I have something missed? would you please have a check on this test? @JChanceHud

**Comment by JChanceHud** [comment]
@wanseob I built a new function into the `BurnAuction` interface to access a `Bid` tuple. I'm not sure how to make a function in an interface return a struct. I made the `highestBidPerRound` state variable public again and the tests should pass now.

**Comment by wanseob** [comment]
I've checkedt that it works very well :+1: . Would you please check this lint result?
image

**Comment by JChanceHud** [comment]
@wanseob Linting issues should be fixed. What do you think of moving the tests from the precommit hook to a CI task? I skip the precommit hook because the tests take too long/sometimes don't work.

**Comment by wanseob** [comment]
> @wanseob Linting issues should be fixed. What do you think of moving the tests from the precommit hook to a CI task? I skip the precommit hook because the tests take too long/sometimes don't work.

I think adding those to the CI task is a good idea but let's keep them in the precommit process because we can easily skip them using `git commit -n` option :)

**Comment by wanseob** [comment]
I'm trying to resolve the `Error: connection not open on send()` & unresolved promise subscriptions. Will make a PR to your branch soon :)

**Pull request merged**